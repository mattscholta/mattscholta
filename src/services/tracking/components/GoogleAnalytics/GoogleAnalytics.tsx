import * as React from 'react';

export interface GoogleAnalyticsProps {
  id: string;
}

/**
 * @name GoogleAnalytics
 * @external https://tagmanager.google.com/
 * @description GTM requires a two part implementation, this script is
 * responsible for loading GTM which loads our Tags and Pixels
 */
export const GoogleAnalytics = (props: GoogleAnalyticsProps) => {
  const { id } = props;

  // Setup
  const __html = `
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());
    gtag('config', '${id}');
  `;

  return <script dangerouslySetInnerHTML={{ __html }} type="text/javascript" />;
};
