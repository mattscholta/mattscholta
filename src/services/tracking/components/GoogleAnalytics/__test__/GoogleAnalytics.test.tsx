import * as React from 'react';
import { render, RenderResult } from '@testing-library/react';

import { GoogleAnalytics, GoogleAnalyticsProps } from '../GoogleAnalytics';

describe('GoogleAnalytics', () => {
  let props: GoogleAnalyticsProps;
  let wrapper: RenderResult;

  const createWrapper = () => <GoogleAnalytics {...props} />;

  beforeEach(() => {
    props = { id: 'test-id' };
    wrapper = render(createWrapper());
  });

  it('matches snapshot', () => {
    expect(wrapper.container.childNodes).toMatchSnapshot();
  });
});
