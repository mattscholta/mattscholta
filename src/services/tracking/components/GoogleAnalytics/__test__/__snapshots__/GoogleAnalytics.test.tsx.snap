// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleAnalytics matches snapshot 1`] = `
NodeList [
  <script
    type="text/javascript"
  >
    
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());
    gtag('config', 'test-id');
  
  </script>,
]
`;
